test/libethereum/blockchain.cpp

```diff
@@ -214,9 +214,12 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 					uncleStream.appendRaw(uncleRlp.out());
 				}
 
-				// update unclehash in case of invalid uncles
-				current_BlockHeader.sha3Uncles = sha3(uncleStream.out());
-				updatePoW(current_BlockHeader);
+				if (vBiUncles.size())
+				{
+					// update unclehash in case of invalid uncles
+					current_BlockHeader.sha3Uncles = sha3(uncleStream.out());
+					updatePoW(current_BlockHeader);
```

Comment:
updatePoW was always called in previous version. Does it no longer need to be called?